Makefile

```diff
@@ -10,7 +10,7 @@ build:
 	go build -o osbuild-upload-aws ./cmd/osbuild-upload-aws/
 	go build -o osbuild-tests ./cmd/osbuild-tests/
 	go build -o osbuild-weldr-tests ./cmd/osbuild-weldr-tests/
-	go build -o osbuild-dnf-json-tests ./cmd/osbuild-dnf-json-tests/
+	go test -c -o osbuild-dnf-json-tests ./cmd/osbuild-dnf-json-tests/
```

Comment:
Why to change only osbuild-dnf-json-tests to use something different then the other testing executables in `/cmd`?

Reply:
With this change `osbuild-dnf-json-tests` is now relying on Go's testing module (notice there is no main function). 

That's POC b/c this is the smallest file. If the approach is well received all of the other executables should be converted in this way as well.